src/pybind/ceph_argparse.py

```diff
@@ -338,8 +338,12 @@ def __init__(self, goodchars=''):
         self.goodset = frozenset(
             [c for c in printable if re.match(goodchars, c)]
         )
+        self.allowempty = allowempty in (True, 'True', 'true')
 
     def valid(self, s: str, partial: bool = False) -> None:
+        if not self.allowempty and s == "":
```

Comment:
We should handle `s == ""` and `s == "  "` both here only using `strip()`.

Reply:
Its been a while ... but I think the whitespace characters are taken care of by the goodchars set definition and python arg parsing. So that condition doesn't need to be tested or handled specifically.
You could give this PR a try and let me know.

The goodchars set is defined at `src/pybind/mgr/volumes/module.py:13` as:
```
goodchars = '[A-Za-z0-9-_.]'
```

Reply:
Yup, I agree . But in that case we will be getting error something like `Invalid command: invalid chars   in  
fs subvolume create <vol_name> <sub_name>` which doesn't seem good. Its better to catch it here and give error as `argument can't be empty`

Reply:
If the input is just spaces (non-empty) and we report as `argument can't be empty`, wouldn't that seem a bit odd ?
I suggest we not overdo things and let the `goodchars` parsing take care of bad chars, including spaces. This whitespace string parsing issue is system-wide. Lets not trip over ourselves.

Reply:
Also, if the input has **some** space chars, the issue of identifying the bad chars still remains.

Reply:
`invalid chars   in   ` the spaces after chars and in is quite odd, its like `something in something` , that's why i was saying better to catch it there and say `argument can't be empty or whitespaces`. But its not a very big issue, we can leave this suggestion:)